src/useTheme.ts

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2024 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { useLayoutEffect, useState } from "react";
+
+import { useUrlParams } from "./UrlParams";
+
+export const useTheme = (): void => {
+  const { theme: themeName } = useUrlParams();
+  const [previousTheme, setPreviousTheme] = useState<string | null>(
```

Comment:
This could be a ref rather than a state variable, which would help us avoid the extra render whenever `previousTheme` is changed